packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -483,6 +489,19 @@ export function createTextNode(
   );
 }
 
+export function resetProperties(
+  domElement: Element,
+  tag: string,
+  rawProps: Object,
+  rootContainerElement: Element | Document | DocumentFragment,
+): void {
+  const attributes = domElement.attributes;
+  while (attributes.length) {
+    domElement.removeAttribute(attributes[0].name);
+  }
+  setInitialProperties(domElement, tag, rawProps, rootContainerElement);
+}
+
```

Comment:
This concept maybe needs more refinement. What we want is to put the already existing node into a state in which it would have been constructed fresh from React. At the moment this does not deal with listeners including ones set by React on previous Placements.

Practically it isn't common to do many placements for html, head, or body. The most common reasons we'd get here are

1) failed hydration falling back to client render
2) client only apps that bind to the document (rare still probably)